testcrate/build.rs

```diff
@@ -37,13 +37,10 @@ fn main() {
         features.insert(Feature::NoSysF16F128Convert);
     }
 
-    if target.starts_with("i586") || target.starts_with("i686") {
+    if target.starts_with("i586") {
         // 32-bit x86 seems to not have `__fixunstfti`, but does have everything else
```

Comment:
I think this is fixed for i586 as well now?

Reply:
I just tried it, looks like they are still not available on i586. I updated the note to mention this

Reply:
I still don't understand why this function is available on i686 but not i586: they should be using the exact same docker image and toolchain.

Reply:
Ok, think I figured it out - i586 and i686 are both missing `__fixunstfti` and `__fixtfti`. i686 was just always disabled at https://github.com/rust-lang/compiler-builtins/pull/631/files#diff-05ae0211bd995d6870067db299c32944cf2b743226442889bb07d48232afc2c5L26 because it has a f128 math bug, https://github.com/rust-lang/compiler-builtins/actions/runs/9640552090/job/26584543080?pr=631#step:13:244.

i586 probably also has that bug, but float tests without sse2 have been disabled since at least https://github.com/rust-lang/compiler-builtins/pull/384 https://github.com/rust-lang/compiler-builtins/blob/0ccc1bfb962883e04bcce1bdc6a6a3a6ac2d2bbc/testcrate/tests/addsub.rs#L113-L132